### PR TITLE
.NET Core 3.0 Port

### PR DESCRIPTION
**Pull request by Inzanit**
Starting port to .NET Core 3.0. **Do not merge. This does not yet build and hasn't been run either.**

This introduces a nightly/preview build of Discord.NET - as Core 3.0 support was recently fixed for `IAsyncEnumerable` as per https://github.com/discord-net/Discord.Net/pull/1382.

This is not on the stable Nuget channel yet - packages are pulled from the reintroduced `NuGet.config`.

Several new members of Discord.NET have not been implemented in our Discord.NET abstraction project, as I wanted to get the project to build first.

This also introduces package updates for EntityFramework, which will mean runtime exceptions in a couple of areas.

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+    <packageSources>
+        <add key="Discord.Net Nightlies" value="https://www.myget.org/F/discord-net/api/v3/index.json" />
```
Comment (tannergooding) [inline NuGet.config]: Why `NuGet.config` rather than a `Directory.Build.props` setting `RestoreSources`?

File: Modix/Modix.csproj [hunk 2]
```diff
@@ -12,21 +12,20 @@
...
   </PropertyGroup>
 
   <ItemGroup>
     <Folder Include="wwwroot\" />
   </ItemGroup>
 
   <ItemGroup>
-    <PackageReference Include="AspNet.Security.OAuth.Discord" Version="2.0.1" />
-    <PackageReference Include="Discord.Net.WebSocket" Version="2.1.1" />
+    <PackageReference Include="AspNet.Security.OAuth.Discord" Version="3.0.0" />
```
Comment (tannergooding) [inline Modix/Modix.csproj]: Versions would be easier to handle if they were centrally managed.

You can define these as `<PackageReference Include"Name" />` and then, in a central `Directory.Build.targets`  do `<PackageReference Update="Name" Version="..." />`

This ensures you have one place to update for version upgrades and makes it easy to manage everything at once or to document places that differ.

Comment (Scott-Caldwell) [reply Modix/Modix.csproj]: That's really cool! I didn't know you could do that

Comment (tannergooding) [reply Modix/Modix.csproj]: You can see an example of how I do this in TerraFX here: <https://github.com/terrafx/terrafx/blob/master/Directory.Build.targets>

Some other repos (like dotnet/project-system) do this as well.

Comment (Inzanit) [reply Modix/Modix.csproj]: Didn't know this existed, has been integrated as part of the latest commit, 👍

File: Modix.Bot/Modules/GuildInfoModule.cs [hunk 3]
```diff
@@ -125,14 +125,19 @@ public async Task AppendGuildParticipationAsync(StringBuilder stringBuilder, Soc
...
 
             stringBuilder
                 .AppendLine(Format.Bold("\u276F Guild Participation"))
                 .AppendLine($"Last 7 days: {"message".ToQuantity(weekTotal, "n0")}")
                 .AppendLine($"Last 30 days: {"message".ToQuantity(monthTotal, "n0")}")
-                .AppendLine($"Avg. per day: {"message".ToQuantity(monthTotal / 30, "n0")}")
-                .AppendLine($"Most active channel: {MentionUtils.MentionChannel(mostActiveChannel.Key)} ({"message".ToQuantity(mostActiveChannel.Value, "n0")} in 30 days)");
+                .AppendLine($"Avg. per day: {"message".ToQuantity(monthTotal / 30, "n0")}");
+
+            if (!(mostActiveChannel is { }))
```
[the remark this answers is not in the document]
Comment (Inzanit) [reply Modix.Bot/Modules/GuildInfoModule.cs]: Good catch, obviously worked during testing 🙄

**Comment by Scott-Caldwell** [comment]
Like Inzanit mentioned in the server, we should probably wait till Friday evening or Saturday to merge this, so that we can be more available to fix any potential issues

**Pull request merged**